pkg/controller/sidecarset/sidecarset_pod_event_handler.go

```diff
@@ -82,6 +102,9 @@ func (p *enqueueRequestForPod) updatePod(q workqueue.RateLimitingInterface, old,
 		if sidecarSet.Spec.UpdateStrategy.Type == appsv1alpha1.NotUpdateSidecarSetStrategyType {
 			continue
 		}
+		if newPod.DeletionTimestamp != nil {
```

Comment:
I think you should not judge the deletionTimestamp, and use the ObserveUpdated function.

Reply:
ObserveUpdated does not determine if the pod is deleted if the pod is pending deletion

Reply:
If this pod is an older version, there may be problems